File: C++_Practice/LeetCode/0001 Two Sum/0001-two-sum.cpp
//#include <unordered_map>
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        vector<int> output;

        for(int i = 0; i < nums.size(); i++){
            int needed = target - nums[i];
            if (seen.contains(needed)){ // found needed in seen
                output.push_back(i);
                output.push_back(seen.at(needed));
                return output;
            }
            seen.insert({nums[i], i});
        }
        return output;
    }
};

File: Python_Practice/LeetCode/0001 Two Sum/0001-Two-Sum.py
class Solution(object):
    def twoSum(self, nums, target):
        seen = {}
        for i in range(len(nums)):
            #print("seen = ", seen)
            need = target - nums[i]
            #print("need = ", need)
            if need in seen:
                return [i, seen[need]]
            seen[nums[i]] = i

            